precode.go

```diff
@@ -7,58 +7,95 @@ import (
 	"sync"
 )
 
-// Generator генерирует последовательность чисел 1,2,3 и т.д. и
+// Generator генерирует последовательность чисел 1, 2, 3 и т.д. и
 // отправляет их в канал ch. При этом после записи в канал для каждого числа
 // вызывается функция fn. Она служит для подсчёта количества и суммы
 // сгенерированных чисел.
 func Generator(ctx context.Context, ch chan<- int64, fn func(int64)) {
-	// 1. Функция Generator
-	// ...
+	defer close(ch) // закрываем канал при выходе из функции
+	i := int64(1)
+	for {
+		select {
+		case <-ctx.Done():
+			return // выходим из функции, если контекст отменен
+		case ch <- i:
+			fn(i)
+			i++
+		}
+	}
 }
 
 // Worker читает число из канала in и пишет его в канал out.
-func Worker(in <-chan int64, out chan<- int64) {
-	// 2. Функция Worker
-	// ...
+func Worker(in <-chan int64, out chan<- int64, wgStat *sync.WaitGroup) {
```

Comment:
Как я уже писал, тут это не нужно
```suggestion
func Worker(in <-chan int64, out chan<- int64) {
```